mkosi/config.py

```diff
@@ -2514,3 +2529,94 @@ def default(self, obj: Any) -> Any:
         elif isinstance(obj, uuid.UUID):
             return str(obj)
         return json.JSONEncoder.default(self, obj)
+
+
+E = TypeVar("E", bound=StrEnum)
+
+
+def json_type_transformer(refcls: Union[type[MkosiArgs], type[MkosiConfig]]) -> Callable[[str, Any], Any]:
```

Comment:
So I wonder if we should make this a bit more robust and user friendly. If we're going to allow users to pass us json objects as config, it'd be nice if we could give proper errors if we can't parse something instead of a stacktrace.

Reply:
I think that should happen at the point of use, which doesn't exist yet, and not in the type transformer or the constructors. I really don't want to `die` in a constructor (yes, there's `die`s all over the place in `parse_config`) as that makes debugging in a REPL, something I do regularly, harder than it needs to be. :) 

I added a try block, though to reraise a single error type from the ones that I think should cover the ones that might appear here. 

I honestly hope, that we get to 3.11 some day and this can be made much prettier with match statements.